src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -1087,12 +1086,11 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
             database.CompleteAllocations();
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void CheckNotDisposed()
         {
             if (_utf8Json.IsEmpty)
             {
-                throw new ObjectDisposedException(nameof(JsonDocument));
+                ThrowHelper.ThrowObjectDisposedException_Utf8JsonWriter();
```

Comment:
We probably need to either tweak this helper slightly or add an additional one, as its message currently refers to Utf8JsonWriter and this is JsonDocument.